Author: Iustin Pop <iusty@k1024.org>
Date:   Sun Oct 5 11:42:31 2008 +0200

Workaround a python unittest failure

Upstream introduced a change to the python unittests in rev 50, which
makes the test pass on i386 but fail on amd64. Workaround this by always
casting the input value to long, as this will make things consistent on
both platforms (upstream is informed, maybe they will have a better
suggestion).
--- a/python/google/protobuf/internal/decoder_test.py
+++ b/python/google/protobuf/internal/decoder_test.py
@@ -122,7 +122,7 @@
         ['fixed64', decoder.Decoder.ReadFixed64, 0xffffffffffffffff,
         'ReadLittleEndian64', 0xffffffffffffffff],
         ['sfixed32', decoder.Decoder.ReadSFixed32, long(-1),
-         'ReadLittleEndian32', 0xffffffff],
+         'ReadLittleEndian32', long(0xffffffff)],
         ['sfixed64', decoder.Decoder.ReadSFixed64, long(-1),
          'ReadLittleEndian64', 0xffffffffffffffff],
         ['float', decoder.Decoder.ReadFloat, 0.0,
